common/network-common/src/main/java/org/apache/spark/network/client/TransportClient.java

```diff
@@ -287,6 +288,9 @@ public void onFailure(Throwable e) {
 
     try {
       return result.get(timeoutMs, TimeUnit.MILLISECONDS);
+    } catch (TimeoutException e) {
+      logger.warn("RPC {} timed-out", rpcId);
+      throw Throwables.propagate(new SaslTimeoutException(e));
```

Comment:
`RpcTimeoutException` in the core module and this is in common. Will just do:
```
throw Throwables.propagate(e.getCause());
```
to avoid making it too complicated for now.